src/services/game-logic/pieces/King.ts

```diff
@@ -0,0 +1,29 @@
+import { PieceNames } from '../../../enums';
+import { Square } from '../../../models/Square';
+import { Piece } from './Piece';
+
+class King extends Piece {
+	name = PieceNames.KING;
+
+	getPossibleMoves(): Square[] {
+		const positions: Array<number[]> = [
+			[ -1, -1 ],
+			[ 0, -1 ],
+			[ 1, 1 ],
+			[ -1, 0 ],
+			[ 1, 0 ],
+			[ 0, 1 ],
+			[ -1, 1 ],
+			[ 1, -1 ]
+		];
+
+		const possibleMoves: Square[] = positions.map((position) => ({
```

Comment:
Możemy wykorzystać destrukturyzacje tablic:  https://www.freecodecamp.org/news/array-and-object-destructuring-in-javascript/

```suggestion
		const possibleMoves: Square[] = positions.map(([row, column]) => ({
```

Reply:
poprawione